### PR TITLE
Refactor and add example project

### PR DESCRIPTION
**Pull request by DudeWhoCode**
# Resolves
- [AR-1807](https://team-1624093970686.atlassian.net/browse/AR-1807)

Note: The example folder is to just show how can we pass around logger down the chain. Its' just stubs and won't be clean

### CODE REVIEW

File: example/models/transaction_db.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+}
+
+func (t *DBTransaction) getTransactionAmount(userName string) int {
+	// Logic to retrieve txns from DB and assemble.
+	t.log.Info("Getting transaction details", logger.Field{
+		"username": userName,
+	})
+
+	return 20
+}
```
Comment (nitisht) [inline example/models/transaction_db.go:27]: Don't see this being used anywhere. Is there a reason to add this in the example

Comment (DudeWhoCode) [reply example/models/transaction_db.go:27]: The example is just to show how could we pass around logger in a complex project. It's not meant to work, it's just stubs as mentioned in PR description.
Nevertheless I think the reason to add it is just return some int as per function signature.

Comment (DudeWhoCode) [reply example/models/transaction_db.go:27]: The intention of example is only to show how to initialize/pass around logger. If it looks dirty, I can clean it up later.

**Review by makylfang — APPROVED** [review]

File: example/models/transaction_db.go (hunk 1 above)
Comment (DudeWhoCode) [reply example/models/transaction_db.go:27]: Added to backlog incase we need to do it, resolving this and merging for further PRs.

https://team-1624093970686.atlassian.net/browse/AR-1909?atlOrigin=eyJpIjoiY2FhZGQwN2RhMzIwNDJlNDlkYmNiYWM4ZGM0ZTc0MDAiLCJwIjoiaiJ9

**Pull request merged**